src/main/java/com/spikes2212/dashboard/AutoChooser.java

```diff
@@ -0,0 +1,96 @@
+package com.spikes2212.dashboard;
+
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj2.command.Command;
+
+import java.util.HashSet;
+
+/**
+ * A class that implements {@link SendableChooser} in order to make an autonomous command chooser.
+ *
+ * @author Ofri Rosenbaum
+ * @see SendableChooser
+ */
+public class AutoChooser extends SendableChooser<Command> {
+
+    /**
+     * A {@link HashSet} that contains the names of all the options in this {@link SendableChooser}.
+     */
+    private HashSet<String> names = new HashSet<>();
+
+    /**
+     * Creates an autonomous chooser from the given commands, where the first command will be the default option and the
+     * rest of the commands will be the rest of the options.
+     *
+     * <p>All options will have names matching their command's {@link Class#getSimpleName()}. In case of multiple
+     * instances of the same class, numbers will also be added to differentiate between the names.</p>
+     *
+     * @param defaultOption the default command this {@link SendableChooser} will use as the default option
+     * @param options the commands that will be added to this {@link SendableChooser} other than the default command
+     */
+    public AutoChooser(Command defaultOption, Command... options) {
+        String defaultName = defaultOption.getClass().getSimpleName();
+        addName(defaultName);
+        setDefaultOption(defaultName, defaultOption);
+        for (Command option : options) {
+            addOption(option.getClass().getSimpleName(), option);
+        }
+    }
+
+    /**
+     * Creates an autonomous chooser where the first {@link Command} and {@link String} will be used as the default
+     * option, and the rest of the options will be parsed as following: every even index (0,2,4,...) will be a command
+     * and every odd index (1,3,5,...) will be used as the name for the previous command.
+     *
+     * @param defaultOption the default command this {@link SendableChooser} will use as the default option
+     * @param defaultOptionName the name for the default command
+     * @param options the rest of the options, where every even index is a command to be added and every odd index is
+     *               the name for the previous command
+     */
+    public AutoChooser(Command defaultOption, String defaultOptionName, Object... options) {
+        setDefaultOption(defaultOptionName, defaultOption);
+        for (int i = 0; i < options.length - options.length % 2; i++) {
+            Command command;
+            String name;
+            if (options[i] instanceof Command) {
```

Comment:
A-IV\a5

Reply:
By using varargs you will have to use an array

Reply:
convert to list